Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/ProcessKeywords.java

```diff
@@ -0,0 +1,504 @@
+     * @return keywords
+     */
+    private static final Set<String> initSimpleKeywords(String input) {
+        Set<String> tokens = new HashSet<>();
+        for (String term : splitValues(input)) {
+            tokens.add(normalize(term));
+        }
+        return tokens;
+    }
+
+    /**
+     * Generates the search terms by batch. Note that batch title can be
+     * optional. Batch ID is not indexed because ID search is done via the
+     * database.
+     * 
+     * @param batches
+     *            batches containing the process
+     * @return batch search terms
+     */
+    private static final Set<String> initBatchKeywords(Collection<Batch> batches) {
+        if (batches.isEmpty()) {
+            return Collections.emptySet();
+        }
+        Set<String> tokens = new HashSet<>();
+        for (Batch batch : batches) {
+            String optionalTitle = batch.getTitle();
+            if (StringUtils.isNotBlank(optionalTitle)) {
+                tokens.addAll(initSimpleKeywords(optionalTitle));
+            }
+        }
+        return tokens;
+    }
+
+    /**
+     * Generates all search terms by task and pseudo search terms.
+     * 
+     * @param tasks
+     *            tasks for the words to be generated
+     * @return search terms and pseudo search terms
+     */
+    private static final Pair<Set<String>, Set<String>> initTaskKeywords(Collection<Task> tasks) {
+        Set<String> taskKeywords = new HashSet<>();
+        Set<String> taskPseudoKeywords = new HashSet<>();
+        for (Task task : tasks) {
+            for (String token : splitValues(task.getTitle())) {
+                String term = normalize(token);
+                taskKeywords.add(term);
+                if (task.isTypeAutomatic()) {
+                    taskKeywords.add(PSEUDOWORD_TASK_AUTOMATIC + VALUE_SEPARATOR + term);
+                }
+                TaskStatus taskStatus = task.getProcessingStatus();
+                if (Objects.isNull(taskStatus)) {
+                    continue;
+                }
+                if (Objects.equals(taskStatus, TaskStatus.DONE)) {
+                    taskPseudoKeywords.add(PSEUDOWORD_TASK_DONE);
+                    taskPseudoKeywords.add(PSEUDOWORD_TASK_DONE + VALUE_SEPARATOR + term);
+                    User closedUser = task.getProcessingUser();
+                    if (Objects.isNull(closedUser)) {
+                        continue;
+                    }
+                    if (StringUtils.isNotBlank(closedUser.getName())) {
+                        taskPseudoKeywords.add(PSEUDOWORD_TASK_DONE_PROCESSING_USER + VALUE_SEPARATOR
+                            + normalize(closedUser.getName()));
+                    }
+                    if (StringUtils.isNotBlank(closedUser.getSurname())) {
+                        taskPseudoKeywords.add(PSEUDOWORD_TASK_DONE_PROCESSING_USER + VALUE_SEPARATOR
+                            + normalize(closedUser.getSurname()));
+                    }
```

Comment:
After a few processes get indexed I noticed in the search data that the name and surname of users which are closed tasks are findable. Is this legal? This should - in my opinion - configurable if this important data (user name and surname) get indexed or not.

Reply:
`stepdoenuser` (`abgeschlossenerschrittbenutzer`) is a choice option in extended search. I don’t know neither if it is needed nor if it is legal.